app/services/payout/potential_payment/uphold_service.rb

```diff
@@ -28,9 +29,9 @@ def perform
     total_probi = probi
 
     # Create the referral payment for the owner
-    unless should_only_notify?
-      PotentialPayment.create(
-        payout_report_id: @payout_report.id,
+    if @publisher.may_register_promo?
```

Comment:
I think we're missing the case where you can no longer register but your already active promo codes have not yet expired

Reply:
Agreed 👍  I added it here along with some tests

https://github.com/brave-intl/publishers/pull/2733/commits/c5add56e31a711592192fe6a08cb7463b1b43afa